.cicd/eosio-contracts-build.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+set -eo pipefail
+. ./.cicd/helpers/general.sh
+CDT_DIR_HOST=$(pwd)
+mkdir -p build_eosio_contracts 
+
+
+[[ ! -z "$CONTRACTS_VERSION" ]] || export CONTRACTS_VERSION="$(cat "$PIPELINE_CONFIG" | jq -r '.dependencies["eosio.contracts"]')"
+git clone -b "$CONTRACTS_VERSION" https://github.com/EOSIO/eosio.contracts.git 
+
+if [[ $(uname) == 'Darwin' ]]; then
+    export PATH=$CDT_DIR_HOST/build/bin:$PATH
+    cd build_eosio_contracts
+    CMAKE="cmake ../eosio.contracts"
+    echo "$ $CMAKE"
+    eval $CMAKE
+    MAKE="make -j $JOBS"
+    echo "$ $MAKE"
+    eval $MAKE
+    cd ..
+else #Linux
+    ARGS=${ARGS:-"--rm --init -v $(pwd):$MOUNTED_DIR"}
+    . $HELPERS_DIR/docker-hash.sh
+
+    PRE_CONTRACTS_COMMAND="export PATH=$MOUNTED_DIR/build/bin:$PATH && cd $MOUNTED_DIR/build_eosio_contracts"
+    BUILD_CONTRACTS_COMMAND="cmake ../eosio.contracts && make -j$JOBS"
+
+    # Docker Commands
+    # Generate Base Images
+    $CICD_DIR/generate-base-images.sh
+    if [[ "$IMAGE_TAG" == 'ubuntu-18.04' ]]; then
+        FULL_TAG='eosio/ci-contracts-builder:base-ubuntu-18.04-develop'
+        export CMAKE_FRAMEWORK_PATH="$MOUNTED_DIR/build:${CMAKE_FRAMEWORK_PATH}"
+        BUILD_CONTRACTS_COMMAND="cmake -DBUILD_TESTS=true $MOUNTED_DIR/eosio.contracts && make -j$JOBS"
+    fi
+
+    COMMANDS_EOSIO_CONTRACTS="$PRE_CONTRACTS_COMMAND && $BUILD_CONTRACTS_COMMAND"
+
+    # Load BUILDKITE Environment Variables for use in docker run
+    if [[ -f $BUILDKITE_ENV_FILE ]]; then
+        evars=""
+        while read -r var; do
+            evars="$evars --env ${var%%=*}"
+        done < "$BUILDKITE_ENV_FILE"
+    fi
+
+    DOCKER_RUN="docker run $ARGS $evars $FULL_TAG bash -c \"$COMMANDS_EOSIO_CONTRACTS\""
+    echo "$ $DOCKER_RUN"
+    eval $DOCKER_RUN
+
+fi
+
+touch wasm_abi_size.json
+cd build/tests/unit/test_contracts
+JSON=$(echo '{}' | jq -r '.')
```

Comment:
**Optional:** The `-r` flag in `jq` here doesn't make sense. That flag removes quotes from string outputs, but we don't have strings here. You should remove it or replace it with `-c`, which means "compact" and prints the JSON with no whitespace. This is desirable when you are storing JSON in BASH variables because it prevents mistakes with BASH looking for inter-frame spacing in some contexts when you do or don't use quotes around variable names.

This unnecessary flag might be my fault, as I gave you some code samples that may have had it by accident.

Reply:
Done.